### PR TITLE
feat(nuxt): add warning in dev mode if layouts/pages do not have a single root node

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/framework/issues/4598, resolves https://github.com/nuxt/framework/issues/6476

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This is a persistent issue for users and this change aims to improve dx in dev mode by proactively warning about components that render a comment (ie. a fragment).

~~@antfu Can you think of a better way of checking this than inspecting the el with a `setTimeout`?~~

~~**Note**: this will only provide the warning on first render at the moment, rather than on subsequent navigation. Will need to look into how that can be done.~~

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by antfu** [comment]
This is the method I could think of as well. Would `nextTick`, or `onMounted` be better than `setTimeout`?

**Comment by danielroe** [comment]
I couldn't think of a way to access `onMounted` within the render function but feel free to implement if you have an idea - it would be more idiomatic (except note https://github.com/nuxt/framework/issues/3587 - 🤦)!

**Comment by danielroe** [comment]
This will only provide the warning on first render at the moment, rather than on subsequent navigation. Will need to look into how that can be done. (Marking as draft for that reason.)

File: packages/nuxt/src/app/components/layout.ts [hunk 1]
```diff
@@ -16,6 +16,18 @@ export default defineComponent({
   setup (props, context) {
     const route = useRoute()
 
+    let vnode: VNode
+    let _layout: string | false
+    if (process.dev && process.client) {
+      onMounted(() => {
```
Comment (pi0) [inline packages/nuxt/src/app/components/layout.ts:22]: Can we somehow expose it as internal util to share with pages?

Comment (danielroe) [reply packages/nuxt/src/app/components/layout.ts:22]: Hm. I've extracted it out to test, and it results in a net increase in LOC (+10) to handle the edge cases (e.g. page message depends on vnode, whereas layout has to be set. It's also a little trickier to ensure it's tree-shaken. But doable if you think the benefits are worth it.

(Personally I would wait until we use it somewhere else before creating shared util.)

**Review by pi0 — APPROVED** [review]
